src/api/routes/rosetta/construction.ts

```diff
@@ -226,47 +263,69 @@ export function createRosettaConstructionRouter(db: DataStore, chainId: ChainID)
     }
     const txSize: number = options.size;
 
-    const recipientAddress = options.token_transfer_recipient_address;
-    if (options?.decimals !== RosettaConstants.decimals) {
-      res.status(400).json(RosettaErrors[RosettaErrorsTypes.invalidCurrencyDecimals]);
-      return;
+    let response = {} as RosettaConstructionMetadataResponse;
+    switch (options.type) {
+      case 'token_transfer':
+        const recipientAddress = options.token_transfer_recipient_address;
+        if (options?.decimals !== RosettaConstants.decimals) {
+          res.status(400).json(RosettaErrors[RosettaErrorsTypes.invalidCurrencyDecimals]);
+          return;
+        }
+
+        if (recipientAddress == null || !isValidC32Address(recipientAddress)) {
+          res.status(400).json(RosettaErrors[RosettaErrorsTypes.invalidRecipient]);
+          return;
+        }
+        break;
+      case 'stacking':
+        // Getting stacking info
+        const poxInfo = await new StacksCoreRpcClient().getPox();
+        const coreInfo = await new StacksCoreRpcClient().getInfo();
+        const contractInfo = poxInfo.contract_id.split('.');
+        options.contract_address = contractInfo[0];
+        options.contract_name = contractInfo[1];
+        // Adding 3 blocks to provide a buffer for transaction to confirm
```

Comment:
Can you explain a bit on how this number was chosen?

Reply:
Sure. I have found it in you documentation step-by-step guide : https://docs.blockstack.org/understand-stacks/integrate-stacking#step-5-lock-stx-to-stack

Could it be an issue ?

Reply:
I don't have the context on how that number was originally decided, but I'm assuming there is a good reason for it 👍